### PR TITLE
feat(AboutModal): update to match new guidelines v11

### PR DESCRIPTION
**Pull request by glapadre**
Contributes to #1776

Styling and padding updated to match [pal](https://pages.github.ibm.com/cdai-design/pal/patterns/about-modal/usage/). Updated storybook stories to match. 

#### What did you change?
`
packages/cloud-cognitive/src/components/AboutModal/AboutModal.js
packages/cloud-cognitive/src/components/AboutModal/_about-modal.scss
packages/cloud-cognitive/src/components/AboutModal/AboutModal.stories.js
packages/cloud-cognitive/src/components/AboutModal/AboutModal.test.js
packages/cloud-cognitive/src/components/AboutModal/_storybook-styles.scss
packages/cloud-cognitive/src/__tests__/__snapshots__/styles.test.js.snap
examples/carbon-for-ibm-products/AboutModal/src/Example/Example.js
`

#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Comment by elycheea** [comment]
@thefirstartist Can you do a quick review of the updated AboutModal before we merge it? Here’s the link to the [deploy preview](https://deploy-preview-2877--v11-carbon-for-ibm-products.netlify.app/).

**Comment by thefirstartist** [comment]
@elycheea All goods except when viewing it in dark theme, the footer bar shouldn't be in white color.  ([design spec](https://pages.github.ibm.com/cdai-design/pal/patterns/about-modal/usage#anatomy)).
image

File: packages/cloud-cognitive/src/components/AboutModal/AboutModal.js
Comment (elycheea) [inline packages/cloud-cognitive/src/components/AboutModal/AboutModal.js:1]: Since version number is called out in the anatomy and something I think we expect to see consistently, it’s probably worth using a prop instead of injecting it as part of the content.

[Figma](https://www.figma.com/file/GqvE8KdjEfxyS6viCHpX6Q/(v11)-Gray-10---IBM-Products?node-id=9478-404995&t=FCGuSHDztrS3FHvf-0) also shows the version number should use `$text-secondary`.

File: packages/cloud-cognitive/src/components/AboutModal/AboutModal.test.js
Comment (elycheea) [inline packages/cloud-cognitive/src/components/AboutModal/AboutModal.test.js:164]: We should keep version number and update this test since version number is a specific element of the AboutModals.

File: packages/cloud-cognitive/src/components/AboutModal/_about-modal.scss
Comment (elycheea) [inline packages/cloud-cognitive/src/components/AboutModal/_about-modal.scss:65]: I think this border-left should update the token to `$border-strong` or `$border-strong-01`.

File: packages/cloud-cognitive/src/components/AboutModal/AboutModal.js
Comment (elycheea) [inline packages/cloud-cognitive/src/components/AboutModal/AboutModal.js:50]: The new guidance for about modal uses the term `General text` to describe the content that goes here. Thoughts on this naming? I feel like this is equivalent to `description` that we use elsewhere. Alternatively, we could possibly have this just as `children`? 🤔

Comment (elycheea) [reply packages/cloud-cognitive/src/components/AboutModal/AboutModal.js:50]: Actually, it could serve the same purpose as `content` ... which if products are using example content currently, in the new design, it probably _should_ fall under the `General text` from guidance. So I think `content` might be more appropriate.

**Review by elycheea — CHANGES_REQUESTED** [review]
Looks good overall minus a few tokens. Actually, I think we could remove the story for `About modal with dark theme` since we have the theme selector now.

Do have some thoughts about 
- adding `version` prop
- `generalText` v. `description` v. `content`

but we can discuss this further with others.

**Comment by glapadre** [comment]
After some further review, adding the `versionNumber` as a prop makes sense so I updated that. I also decided to change `generalText` to `content` as it is the most generic fitting term for what it represents.

File: packages/ibm-products/src/components/AboutModal/AboutModal.stories.js
Comment (elycheea) [inline packages/ibm-products/src/components/AboutModal/AboutModal.stories.js:201]: We can actually probably remove the `with links` story since the `links` prop in Storybook will allow you to add/remove links to the story.

In fact, most things you should be able to accomplish in a 1-2 stories if you update the controls to be more like links. But anything that would be a bit more involved to update (e.g. different variations of `content`) **can definitely be addressed later** than just removing redundant stories like links or even “all props set.”

**Review by elycheea — COMMENTED** [review]
Looks great! Just had a couple comments on prop names and typing but probably better to discuss with a few others.

Also think we could reduce the stories since _most_ things are doable if you leverage the controls. Some changes to consolidate the stories will probably take a little more time though in which case I’d say to merge this once we resolve the prop discussion and address those when you have more time.

**Review by elycheea — APPROVED** [review]

**Pull request merged**
